tests/framework/test_state_machine.py

```diff
@@ -35,38 +23,120 @@ def test_initialize_allowing_self_transition():
     assert not undefined_self_transitions.transition_set.allow_null_transition
 
 
-def test_transition():
-    done_state = State("done")
+def test_initialize_with_initial_state() -> None:
     start_state = State("start")
-    start_state.add_transition(Transition(start_state, done_state))
-    machine = Machine("state", states=[start_state, done_state])
+    other_state = State("other")
+    machine = Machine("state", states=[start_state, other_state], initial_state=start_state)
+    simulation = InteractiveContext(components=[machine])
+    assert simulation.get_population()["state"].unique() == ["start"]
 
+
+def test_initialize_with_scalar_initialization_weights(
+    base_config: LayeredConfigTree,
+) -> None:
+    base_config.update(
+        {"population": {"population_size": 10000}, "randomness": {"key_columns": []}}
+    )
+    state_a = State("a", initialization_weights=lambda _: 0.2)
+    state_b = State("b", initialization_weights=lambda _: 0.8)
+    machine = Machine("state", states=[state_a, state_b])
+    simulation = InteractiveContext(components=[machine], configuration=base_config)
+
+    state = simulation.get_population()["state"]
+    assert np.all(simulation.get_population().state != "start")
+    assert round((state == "a").mean(), 1) == 0.2
```

Comment:
Any reason not to use `np.isclose`? 0.24 would rown down to 0.2 but is an unacceptable 16% different.

Reply:
Mostly laziness (and it allows for smaller populations. The other tests in this module use `round`. We should really use the FuzzyChecker.